<a href="https://colab.research.google.com/github/PoloVlzq/EXAMEN/blob/main/Problema_19.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Problema 19
Las bombillas eléctricas de un fabricante A tiene una duración media a 1400 horas con
una desviación típica de 200 horas, mientras que las de otro fabricante B tiene una
duración media de 1200 horas con una desviación típica de 100 horas. Si se toman
muestras al azar de 125 bombillas de cada fabricante, ¿cual es la probabilidad de que
las bombillas de A tengan una duración media que sea al menos(a) 160 horas, (b) 250
horas mas quelas bombillas de B?

In [1]:
import random
from random import randrange

Obtenemos los valores aleatorios de la varianza de los 125 casos para la bombilla A y B, modelamos el evento una sola vez:

In [154]:
#tamano de la muestra
m=125
#paso
i=1
#conjunto de duracion de las bombillas A
a=[]
#conjunto de duración de las bombillas B
b=[]
#experimento
while m>i:
  #variable aleatorio para la varianza de A y B
  v1=random.randrange(-200,201,1)
  v2=random.randrange(-100,101,1)
  a.append(1400+v1)
  b.append(1200+v2)
  i=i+1
#paso
i=0
#numero de muestras de A mayores a B por 160 hrs
p=0
#numero de muestras de A mayores a B por 250 hrs
pp=0
while i<124:
  d=a[i]-b[i]
  if d>=160:
    p=p+1
  if d>=250:
    pp=pp+1
  i=i+1
prom=p/m
pprom=pp/m
print(prom)
print(pprom)

0.552
0.392


Procedemos a realizar el experimento numerosas veces más en en conjuntos

In [184]:
#conjunto de promedios para el caso A
pp1=[]
#conjunto de promedios para el caso B
pp2=[]
#cantidad de n conjunto de eventos
nn=50
#paso
iii=0
while nn>iii:
#promedio para el caso A
  p1=[]
#promedio para el caso B
  p2=[]
#cantidad de eventos
  n=1000
#paso
  ii=0
  while n>ii:
    m=125
    i=1
    a=[]
    b=[]
    while m>i:
      v1=random.randrange(-200,201,1)
      v2=random.randrange(-100,101,1)
      a.append(1400+v1)
      b.append(1200+v2)
      i=i+1
    i=0
    p=0
    pp=0
    while i<124:
      d=a[i]-b[i]
      if d>=160:
        p=p+1
      if d>=250:
        pp=pp+1
      i=i+1
    prom=p/m
    pprom=pp/m
    p1.append(prom)
    p2.append(pprom)
    ii=ii+1
  g1=sum(p1)/len(p1)
  g2=sum(p2)/len(p2)
  pp1.append(g1)
  pp2.append(g2)
  iii=iii+1  
print(sum(pp1)/len(pp1)*100,"%")
print(sum(pp2)/len(pp2)*100,"%")


59.61347199999998 %
37.360511999999844 %


Se observa una tendecia de los dato a 60% de probabilidad para el caso donde
las bombillas de A tengan una duración media que sea al menos 160 horas mas que las bombillas de B

Para el segundo caso donde las bombillas de A tengan una duración media que sea al menos 250 horas mas que las bombillas de B el porcentaje tiene a 37% de probabilidad